### ISSUE TITLE
Translations

### ISSUE THREAD

**Issue by OxygenCobalt** [OxygenCobalt/Auxio#3]
Many of Auxio's translations are likely incorrect or incomplete. So this issue thread is meant to pool translations for this project.

#### Finding your country code [A.K.A your `values-x`]

Go to [here](https://www.iso.org/obp/ui/#search). Select "Country Codes" and then search for the name of your country. You should see a 2-letter combination under the `Alpha-2` column. This is your **country code**.
- Usually country codes are all that is needed to determine a language directory, with the general format being `values-x` where x is the country code.
    i. **German** would have a translation directory named `values-de` 
    ii. **Polish** would have a translation directory named of `values-pl`
- Sometimes, languages can differ heavily across regions. If this is the case, format your directory as `values-x-rY`, where x is the code for the country that your language **originated** from and Y is your country code in all uppercase.
   i. **Brazilian Portuguese** would have a translation directory named `values-pt-rBR`
   ii. **Taiwanese Chinese** would have a translation directory named `values-zh-rTW`

#### Translating Strings

1. After determining what country code you should use, download the main [strings.xml](https://github.com/OxygenCobalt/Auxio/blob/dev/app/src/main/res/values/strings.xml) file and open it in your preferred text editor. You could also search for an existing `strings.xml` that corresponds to your country code [here](https://github.com/OxygenCobalt/Auxio/blob/dev/app/src/main/res/values/strings.xml) and edit that, although you will need to add XML tags for new translations.
2. Begin translating. You can ignore the `<string name="XYZ">` and `</string>` bits, only edit the contents in between.
3. [Optional]. If your language has more/less than two ways of describing the quantities of something, make sure to modify the `<plurals>` strings to reflect that, adding/removing categories as needed. You can find more information on modifying plurals to fit your language [here](https://developer.android.com/guide/topics/resources/string-resource.html#Plurals).
4. When you are done with your translations, you can fork Auxio and create a [Pull Request](https://github.com/OxygenCobalt/Auxio/pulls) with your added translations, or you can add your translated `strings.xml` to a reply to this thread for me to add instead, just make sure to add the name of your `values` directory so that I know where to put it.

#### Translating F-Droid Descriptions

1. Download the Auxio codebase to your system. This can be done by either cloning the repository using `git clone` or downloading it directly from `Code` -> `Download ZIP`.
2. Enter the `fastlane/metadata/android` directory, and create a new directory with your **country code** from earlier as the name.
3. Copy over the **text** files from the main `en-US` directory to the new directory. These would be `full_description.txt`, `short_description.txt`, and [Optionally] the changelogs in the `changelogs` directory. Do not copy the images.
4. Begin translating, this should be relatively straightfoward, just ignore all the HTML tags [Such as `<a>`] and focus on the contents themselves.
5. When you are done, you can upload a `zip` file of your translated metadata for me to add, or fork Auxio directly and create a [Pull Request](https://github.com/OxygenCobalt/Auxio/pulls).

I'm here to clarify anything or deal with problems, thanks in advance.

**Comment by Atrate** [comment]
I feel like it would be beneficial to use a translation platform for that, as it speeds up the workflow considerably and would definitely help with maintenance as new strings get added. AFAIK, FOSS projects can use weblate for free (https://hosted.weblate.org/hosting/)

From the website: 
```
Please check in advance whether your project is eligible for gratis project hosting on this site.

    Translated content has to be released under a license approved by OSI or recognized as libre by FSF.
    All the source code has to be publicly available in a supported version control system.
    Translations have to be in the same repository as the source code, or under the same project/organization as the source code repository.
    Using Weblate has been approved by the upstream project.
    Mention you use Weblate for translations in the README. It’ll attract new translators and also help Weblate to be free for more projects.
    The project has had active development for at least three months.
    You will have 14 days to set up your project.
    Please request project approval once you finish the setup.
    Unapproved projects will be removed.
    There is no guarantee of service availability, nor quality for free service plan.
```

**Comment by OxygenCobalt** [comment]
I've considered weblate before, but I don't like the idea of having a component of my project reliant on some server far away [Which is what hosted weblate does]. I would want to self-host it before I could implement this.

**Comment by lisiczka43** [comment]
updated russian translation (ru), the one that came with the latest update was extremely amateurish, and there's no option to switch the app to english, so i was forced to create github account and fix it myself (lol). idk how to make a pull request and all that stuff, so i attach it here, hopefuly @oxygencobalt you'll see it

changes to tanslation include: using less literal translations, using shorter wording to fit into UI elements, correcting mistranslations, improving consistency of terminolgy, enforcing strict ё, adding missing new strings

[strings.zip](https://github.com/OxygenCobalt/Auxio/files/8006870/strings.zip)

**Comment by OxygenCobalt** [comment]
> updated russian translation (ru)
Thanks for that. I'll try to implement it as soon as I am done with this refactor I am working on.

> and there's no option to switch the app to english
I'm going to try to implement that in a future version. Seems like a lot of people have use for it.

**Comment by lisiczka43** [comment]
>I'm going to try to implement that in a future version. Seems like a lot of people have use for it.

appreciate it my man, i'm also looking forward for roboto migration, not really a fan of new google fonts

**Comment by OxygenCobalt** [comment]
Oh, about that. It's not so much the google font as it is a font called [inter](https://github.com/rsms/inter). 

Why don't you like the font? I've been unsure of whether to actually commit to the migration for two reasons:
- I generally prefer Inter's bolded text over Roboto's, and I can't just use both
- Inter distinguishes Auxio a bit more from other apps that use Roboto everywhere

Perhaps I can add a configuration option that disables the custom font.

**Comment by lisiczka43** [comment]
hmm, i though this font was a part of material you redesign. anyway, that's just my preference when apps use system fonts. nothing serious really. i was checking out commits and decided to comment on it. if you like the font you should keep it

**Comment by voxrie** [comment]
Hey, can you create a project on oneskyapp or crowdin? That way would be more easy to translate.

**Comment by OxygenCobalt** [comment]
Okay, I'll consider adding weblate. My previous objections somewhat stemmed from the relative obscurity of Auxio, but now with the amount of people using my app, I may as well apply for FOSS weblate.

**Comment by OxygenCobalt** [comment]
Closing this thread in favor of Auxio's Weblate project at https://hosted.weblate.org/engage/auxio/.

@ktutsch @cccClyde @Fjuro @Gsset @enricocid @n-berenice if you prefer, you can now submit new translations via Weblate instead of making a pull request.

**Comment by enricocid** [comment]
Super!
Strings like 
%d kbps or audio formats can be safely moved to donottranslate, I think :)
-

**Comment by OxygenCobalt** [comment]
I don't think so. `kbps` could feasibly be written differently in other alphabets (Cyrillic and Arabic come to mind). For audio formats, since there are still translatable bits (Most often the "Audio" part), I decided to make them translatable.

**Comment by ktutsch** [comment]
I translated the "Audio" from the formats in the German translation. When looking up the Wikipedia article for data rate units (https://en.wikipedia.org/wiki/Data-rate_units), and switching to a language with a different alphabet, everything was written with that alphabet. I don't know these languages, but I seems like the units are translatable.

**Comment by OxygenCobalt** [comment]
Yeah, also, in the Android OS itself, size units (which are very similar to bitrate units) also [https://cs.android.com/search?q=kilobyteShort&ss=android%2Fplatform%2Fsuperproject](have different translations), so I find it reasonable to make it translatable.

**Comment by n-berenice** [comment]
Thanks for the update. Already following the project, hopefully it'll be easier than PRs since I'm not a dev.

**Comment by OxygenCobalt** [comment]
Just realized that Weblate was not synced with the Auxio repo. Should be fixed now. @ktutsch @cccClyde @Fjuro @Gsset @enricocid @n-berenice

**Comment by enricocid** [comment]
It would be nice if you could use GitHub Actions to post debug builds. I find it hard to translate without context 😁

**Comment by OxygenCobalt** [comment]
@enricocid Sorry to necro, but I've set up github actions now, so you can translate with context.

**Comment by tmgldn** [comment]
Hey @OxygenCobalt - was hoping to add en_GB via Weblate, but Weblate requires the maintainer to manually enable English (UK) as a language option - please would you be able to do this when you get a chance?
